python-frontend/src/main/java/org/sonar/plugins/python/api/types/InferredType.java

```diff
@@ -44,4 +44,7 @@
   @Beta
   boolean isCompatibleWith(InferredType other);
 
+  @Beta
+  Optional<Symbol> declaringSymbol();
```

Comment:
Within ticket SONARPY-728 (https://github.com/SonarSource/sonar-python/pull/789), we're adding 
```java
Collection<Symbol> declaringSymbols();
``` 
to handle cases where we have more than one (i.e. union types).
Could you please rebase as soon as it will be merged and use that API instead?

Reply:
After discussion, we decided not to expose the "declaring Symbol" of an Inferred Type. 
We will instead expose "type hints" from FunctionSymbol